lib/foreman_maintain/concerns/downstream.rb

```diff
@@ -74,21 +74,36 @@ def rh_repos(server_version)
 
       def ansible_repo(server_version)
         if server_version >= version('6.8')
-          "rhel-#{el_major_version}-server-ansible-2.9-rpms"
+          ansible_version = '2.9'
         elsif server_version >= version('6.6')
-          "rhel-#{el_major_version}-server-ansible-2.8-rpms"
+          ansible_version = '2.8'
         elsif server_version >= version('6.4')
-          "rhel-#{el_major_version}-server-ansible-2.6-rpms"
+          ansible_version = '2.6'
         end
+
+        if el7?
+          "rhel-#{el_major_version}-server-ansible-#{ansible_version}-rpms"
+        else
+          "ansible-#{ansible_version}-for-rhel-#{el_major_version}-x86_64-rpms"
+        end
+      end
+
+      def use_beta_repos?
+        ENV['FOREMAN_MAINTAIN_USE_BETA'] == '1'
       end
 
       def product_specific_repos(full_version)
-        repos = []
-        repos << if ENV['FOREMAN_MAINTAIN_USE_BETA'] == '1'
-                   "rhel-server-#{el_major_version}-#{package_name}-6-beta-rpms"
-                 else
-                   "rhel-#{el_major_version}-server-#{package_name}-#{full_version}-rpms"
-                 end
+        maj_version = full_version[0]
+        repos = if el7? && use_beta_repos?
+                  ["rhel-server-#{el_major_version}-#{package_name}-#{maj_version}-beta-rpms"]
+                elsif el7?
+                  ["rhel-#{el_major_version}-server-#{package_name}-#{full_version}-rpms"]
+                elsif use_beta_repos?
+                  ["#{package_name}-#{maj_version}-beta-for-rhel-#{el_major_version}-x86_64-rpms"]
```

Comment:
this follows the style of `satellite-tools-6-beta-for-rhel-8-x86_64-rpms` which we already have today.